### PR TITLE
refactor(core): move commonAncestorLookup to TileMesh

### PR DESCRIPTION
**Pull request by zarov**
Move it from all the views to the prototype of TileMesh.

### CODE REVIEW

**Review by peppsac — COMMENTED** [review]
Could you add a couple of unit-tests for this please?

Also the newly introduced method should start with a verb ( `findLowestCommunAncestor`?)

**Comment by zarov** [comment]
No problem, doing it.

**Comment by zarov** [comment]
Updated

File: src/Core/TileMesh.js [hunk 1]
```diff
@@ -237,4 +237,31 @@ TileMesh.prototype.getZoomForLayer = function getZoomForLayer(layer) {
...
 
+/**
+ * Search for a common ancestor between this tile and another one. It goes
+ * through parents on each side until one is found.
+ *
+ * @param {TileMesh} tile
+ *
+ * @return {TileMesh} the resulting common ancestor
+ */
+TileMesh.prototype.findCommonAncestor = function findCommonAncestor(tile) {
```
Comment (gchoqueux) [inline src/Core/TileMesh.js:248]: is that [this code](https://github.com/iTowns/itowns/pull/719/files#diff-6b60bae2240ab556c955b4eb0a07a12eR189)  is similar in ancestor search with different criteria (id and name)

Comment (zarov) [reply src/Core/TileMesh.js:248]: It is similar, but I don't see how it could be mutualized: pointcloud is a very specific usecase.

**Review by peppsac — APPROVED** [review]
LGTM, thanks!

**Pull request merged**